inc/knowbaseitem.class.php

```diff
@@ -957,6 +957,57 @@ function showFull($options = []) {
       $out.= "</div>";
       $out.= "</td></tr>";
 
+      // Show documents attached to the FAQ Item
+      $sort = 'name';
+      $order = 'ASC';
+      $criteria = Document_Item::getDocumentForItemRequest($this, ["$sort $order"]);
+      $iterator = $DB->request($criteria);
+      if (count($iterator) > 0) {
+         $out.= "<tr><td class='left' colspan='4'><h2>".Document::getTypeName(Session::getPluralNumber())."</h2></td></tr>\n";
+
+         $columns = [
+            'name'      => __('Name'),
+            'filename'  => __('File'),
+            'headings'  => __('Heading'),
+            'assocdate' => _n('Date', 'Dates', 1),
+         ];
+
+         $header_begin  = "<tr>";
+         $header_top    = '';
+         $header_end    = '';
+
+         foreach ($columns as $key => $val) {
+            $header_end .= "<th".($sort == "$key" ? " class='order_$order'" : '').">".
+                           "<a href='javascript:reloadTab(\"sort=$key&amp;order=".
+                              (($order == "ASC") ?"DESC":"ASC")."&amp;start=0\");'>$val</a></th>";
+         }
+         $header_end .= "</tr>";
+         $out .= $header_begin.$header_top.$header_end;
+
+         $document = new Document();
+         foreach ($iterator as $data) {
+            $docID        = $data["id"];
+            $link         = NOT_AVAILABLE;
+            $downloadlink = NOT_AVAILABLE;
+
+            if ($document->getFromDB($docID)) {
+               $link         = $document->getLink();
```

Comment:
is, from a self-service pov, required to have a link to the glpi document.
I think the targeted page is not available to this kind of profiles

Reply:
I did a test with a self service account. I can access a .png file from the link.

![image](https://user-images.githubusercontent.com/14139801/111765539-c2413700-88a4-11eb-979c-a84cf204a58d.png)


![image](https://user-images.githubusercontent.com/14139801/111765571-cf5e2600-88a4-11eb-82bc-3d0287111b49.png)

Reply:
No, i talk about the document link not the download link.

But last week we also talked about having a name and a file name is redundant
Please show the link as
```
<a href="download_link">document name</a>
```

Reply:
redundancy removed, also filtered out deleted files.